### PR TITLE
Update docker config for OIDC gcr failure

### PR DESCRIPTION
**Pull request by Aishwarya-Lad**
Change the login to docker login command instead of using the Config struct for OIDC changes.
 Made the decision to use login instead of writing file directly.

What is tested- GCR/ECR/ACR build and push steps with DLC
https://docs.google.com/document/d/1dzLf7PR43NiTjdScxu5gUgWO8O3QTonaDupSDIivFuU/edit

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -178,22 +178,22 @@ func (p Plugin) Exec() error {
...
-		}
-		defer file.Close()
-		_, err = file.Write(json)
+		cmd := exec.Command(
+			dockerExe, "login",
+			"-u", p.BaseImageUsername,
+			"-p", p.BaseImagePassword,
+			p.BaseImageRegistry,
+		)
+		fmt.Println(" Login Command is : ", cmd)
```
Comment (jamie-harness) [inline docker.go:187]: remove this line

**Review by jamie-harness — CHANGES_REQUESTED** [review]

File: docker.go [hunk 2]
```diff
@@ -178,22 +178,22 @@ func (p Plugin) Exec() error {
...
-		}
-		os.MkdirAll(dockerHome, 0600)
-		path := filepath.Join(dockerHome, "config.json")
-		file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
-		if err != nil {
-			return fmt.Errorf("Error opening config.json: %s", err)
-		}
-		defer file.Close()
-		_, err = file.Write(json)
+		cmd := exec.Command(
```
Comment (jamie-harness) [inline docker.go:181]: refactor this to use `commandLogin`. Create a new login from baseImage credentials

**Review by jamie-harness — CHANGES_REQUESTED** [review]

File: docker.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Aishwarya-Lad) [reply docker.go]: Yes, removed it.

File: docker.go (hunk 2 above)
Comment (Aishwarya-Lad) [reply docker.go]: Done.

File: app.go [hunk 3]
```diff
@@ -364,11 +364,12 @@ func run(c *cli.Context) error {
...
-			Username: c.String("docker.username"),
-			Password: c.String("docker.password"),
-			Email:    c.String("docker.email"),
-			Config:   c.String("docker.config"),
+			Registry: 	 c.String("docker.registry"),
+			Username:    c.String("docker.username"),
+			Password: 	 c.String("docker.password"),
+			Email:    	 c.String("docker.email"),
+			Config:   	 c.String("docker.config"),
+			AccessToken: c.String("access-token"),
```
Comment (Ompragash) [inline app.go:372]: just run "go fmt" once on the changed files. Rest LGTM!

File: docker.go [hunk 4]
```diff
@@ -177,23 +177,23 @@ func (p Plugin) Exec() error {
 	}
 
 	// add base image docker credentials to the existing config file, else create new
+	// instead of writing to config file directly, using docker's login func
```
Comment (hemanthmantri) [inline docker.go:180]: can you also explain what's the difference and why one is preferred to another?

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: docker.go [hunk 5]
```diff
@@ -177,23 +177,23 @@ func (p Plugin) Exec() error {
...
+		baseConnectorLogin.Password = p.BaseImagePassword
+
+		cmd := commandLogin(baseConnectorLogin)
+
+		raw, err := cmd.CombinedOutput()
 		if err != nil {
-			return fmt.Errorf("Error writing config.json: %s", err)
+			out := string(raw)
+			out = strings.Replace(out, "WARNING! Using --password via the CLI is insecure. Use --password-stdin.", "", -1)
+			fmt.Println(out)
```
Comment (hemanthmantri) [inline docker.go:193]: What is this error message? If this is printed what's the change the customer has to do?

File: go.mod [hunk 6]
```diff
@@ -15,12 +16,21 @@ require (
...
+	github.com/opencontainers/image-spec v1.1.0 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
-	golang.org/x/sys v0.0.0-20220731174439-a90be440212d // indirect
+	golang.org/x/sys v0.1.0 // indirect
 	gopkg.in/yaml.v2 v2.2.8 // indirect
-	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	gotest.tools/v3 v3.5.1 // indirect
```
Comment (hemanthmantri) [inline go.mod:33]: It seems a lot of packages got added. Is this expected?

File: docker.go (hunk 5 above)
Comment (Aishwarya-Lad) [reply docker.go:193]: just replacing the general docker warning, not an error.

File: docker.go (hunk 4 above)
Comment (Aishwarya-Lad) [reply docker.go:180]: The login function integrates with various credential helpers, it also handles different registry specific logic.

File: go.mod (hunk 6 above)
Comment (Aishwarya-Lad) [reply go.mod]: modified

**Review by jamie-harness — APPROVED** [review]

**Review by Ompragash — APPROVED** [review]

**Pull request merged**